source/zarchive.d

```diff
@@ -236,6 +238,50 @@ public:
         return false;
     }
 
+    void extractArchive (const string dest)
+    in
+    {
+        import std.file : isDir;
+        assert (dest.isDir);
+    }
+    body
+    {
+        import std.file : chdir, getcwd;
+
+        archive *ar;
+        archive *ar_dest;
+        archive_entry *en;
+
+        const void *buf;
+        size_t size;
+        long offset;
+
+        string cwd;
+
+        cwd = getcwd ();
+
+        ar = openArchive ();
+        scope(exit) archive_read_free (ar);
+
+        dest.chdir;
+        scope(exit) cwd.chdir;
```

Comment:
libarchive's API always extracts to the working directory. There's some thread safety problems here if things use relative paths. I only found one problem there in practice, and made it use an absolute path.